File: runes/rune_system.py
"""
Sistema de Runas – Valhalla Rising – The Parchment
Desenvolvido por Marcelo (2025)
Linguagem: Python 3
"""

# Estrutura básica para combinação e avaliação de runas

class Runa:
    def __init__(self, nome, cultura, subcategoria, raridade, tipo, sinergia):
        self.nome = nome
        self.cultura = cultura
        self.subcategoria = subcategoria
        self.raridade = raridade
        self.tipo = tipo
        self.sinergia = sinergia  # Valor de 1 a 3

class BuildRunica:
    def __init__(self):
        self.runas = []

    def adicionar_runa(self, runa):
        if len(self.runas) < 3 and not self.tem_cultura(runa.cultura):
            self.runas.append(runa)
            return True
        return False

    def tem_cultura(self, cultura):
        return any(r.cultura == cultura for r in self.runas)

    def eficiencia_total(self):
        return sum(r.sinergia for r in self.runas)

# Exemplo de uso
if __name__ == "__main__":
    r1 = Runa("Espinho de Garmr", "Viking", "Força Adaptiva", "Rara", "Condicional", 2)
    r2 = Runa("Lança de Tezcatlipoca", "Maya", "Foco", "Épica", "Condicional", 3)
    r3 = Runa("Sandálias de Hermes", "Grega", "Velocidade", "Comum", "Passiva", 2)

    build = BuildRunica()
    build.adicionar_runa(r1)
    build.adicionar_runa(r2)
    build.adicionar_runa(r3)

    print("Eficiência total da build:", build.eficiencia_total())

File: runes/RuneSystem.java
// Sistema de Runas – Valhalla Rising – The Parchment
// Desenvolvido por Marcelo (2025)

public class Runa {
    String nome, cultura, subcategoria, raridade, tipo;
    int sinergia;

    public Runa(String nome, String cultura, String subcategoria, String raridade, String tipo, int sinergia) {
        this.nome = nome;
        this.cultura = cultura;
        this.subcategoria = subcategoria;
        this.raridade = raridade;
        this.tipo = tipo;
        this.sinergia = sinergia;
    }
}

import java.util.ArrayList;

public class BuildRunica {
    ArrayList<Runa> runas = new ArrayList<>();

    public boolean adicionarRuna(Runa r) {
        if (runas.size() < 3 && !temCultura(r.cultura)) {
            runas.add(r);
            return true;
        }
        return false;
    }

    public boolean temCultura(String cultura) {
        for (Runa r : runas) {
            if (r.cultura.equals(cultura)) return true;
        }
        return false;
    }

    public int eficienciaTotal() {
        int total = 0;
        for (Runa r : runas) {
            total += r.sinergia;
        }
        return total;
    }

    public static void main(String[] args) {
        Runa r1 = new Runa("Espinho de Garmr", "Viking", "Força Adaptiva", "Rara", "Condicional", 2);
        Runa r2 = new Runa("Lança de Tezcatlipoca", "Maya", "Foco", "Épica", "Condicional", 3);
        Runa r3 = new Runa("Sandálias de Hermes", "Grega", "Velocidade", "Comum", "Passiva", 2);

        BuildRunica build = new BuildRunica();
        build.adicionarRuna(r1);
        build.adicionarRuna(r2);
        build.adicionarRuna(r3);

        System.out.println("Eficiência total da build: " + build.eficienciaTotal());
    }
}
